In [1]:
import pandas

# Exact String Match

In [2]:
df = pandas.read_pickle("results/NMVWtoWikiExactMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

3360

In [8]:
df.loc[df["match"] == "NO"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,retrieved_names,match
5656,https://hdl.handle.net/20.500.11840/pi80338,A.B. (Ada Bird) Petyarre,http://www.wikidata.org/entity/Q27077624,"[Ada Petyarre Bird, Ada Bird Petyarre, Ada Pet...",[],NO
4495,https://hdl.handle.net/20.500.11840/pi69977,Shi Zichen,http://www.wikidata.org/entity/Q82575085,"[Zichen Shi, Zichen Shi, Zichen Shi]",[],NO
4666,https://hdl.handle.net/20.500.11840/pi76699,"Albert, R.",http://www.wikidata.org/entity/Q83286830,"[R. Albert, R. Albert, R. Albert, R. Albert, R...",[],NO
280,https://hdl.handle.net/20.500.11840/pi22767,J.C. (Jacob Christiaan) Koningsberger,http://www.wikidata.org/entity/Q2292859,"[Jacob Christiaan Koningsberger, Jacob Christi...",[],NO
5436,https://hdl.handle.net/20.500.11840/pi79507,O. Zerkowitz,http://www.wikidata.org/entity/Q83290955,[],[],NO
94,https://hdl.handle.net/20.500.11840/pi20636,J.S.G. (Jan Simon Gerardus) Gramberg,http://www.wikidata.org/entity/Q16062716,"[Jan Simon Gerardus Gramberg, Jan Simon Gerard...",[],NO
886,https://hdl.handle.net/20.500.11840/pi40058,dhr. J. Lie a Fo,http://www.wikidata.org/entity/Q3929328,"[John Lie A Fo, John Lie A Fo, John Lie A Fo, ...",[],NO
316,https://hdl.handle.net/20.500.11840/pi24161,C. (Carl) Bock,http://www.wikidata.org/entity/Q1036746,"[Carl Bock, Carl Bock, كارل بوك, كارل بوك, Car...",[],NO
1917,https://hdl.handle.net/20.500.11840/pi4758,Prof. Dr. J.A. (Jan Antonie) Honing,http://www.wikidata.org/entity/Q63003525,"[Jan Antonie Honing, Jan Antonie Honing, Jan A...",[],NO
5139,https://hdl.handle.net/20.500.11840/pi7817,Dr. A.B. Meyer,http://www.wikidata.org/entity/Q63167,"[أدولف ماير, ادولف ماير, Adolf Bernhard Meyer,...",[],NO


Reason for Mismatch: 
    - Different data keeping practices, it varies depending on who kept the record A.B. (Ada Bird) Petyarre	--> Ada Petyarre Bird; both initial and name in the bracket is present
    - position mismatch for firstname and lastname: Shi Zichen --> Zichen Shi
    - Surname first then abbreviation: Albert, R. --> R. Albert
    - Partial abbreviation: J. Lie a Fo --> John Lie A Fo
    - addressed with Dhr. or Mw or mevr
    - Title, e.g., Prof. Dr. present: Prof. Dr. J.A. (Jan Antonie) Honing --> Jan Antonie Honing
    - The name is mention through abbreviation, not with full name: L. Scherman --> Lucian Scherman
    - Mentioned with just middle name: W. (Wilfrid) Daleus --> Joseph Wilfrid Daleus
    - character encoding went wrong: Böeseken, A.--> A. Böeseken
    - why 997 is not a match
    - why 2323 and 5455 is empty
    - The word junior is mentioned in many different way: Jr. Goemans --> Goemans Jr, J. (John) Cleveley the Younger --> John, Jr. Cleveley, John, Jr. Cleveley,...

# Surname Matching

In [3]:
df = pandas.read_pickle("results/NMVWtoWikiSurnameMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

5245

In [12]:
df.loc[df["match"] == "NO"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,retrieved_names,match
930,https://hdl.handle.net/20.500.11840/pi40328,M. (Metin) Özlen,http://www.wikidata.org/entity/Q82574219,"[Metin Özlen, Metin Özlen, Metin Özlen]",[],NO
2424,https://hdl.handle.net/20.500.11840/pi60258,Shunkôsai Hokushû,http://www.wikidata.org/entity/Q3074677,"[شونكوساى هوكوشو, Shunkōsai Hokushū, Shunkōsai...",[],NO
2389,https://hdl.handle.net/20.500.11840/pi60197,Seki Yûshi,http://www.wikidata.org/entity/Q86818944,"[Seki Yûshi, Seki Yûshi]",[],NO
4950,https://hdl.handle.net/20.500.11840/pi77587,G. Wigand,http://www.wikidata.org/entity/Q84594433,[],[],NO
561,https://hdl.handle.net/20.500.11840/pi38705,Z. Prévost,http://www.wikidata.org/entity/Q50810310,"[Zachée Prévost, Zachée Prévost, Zachée Prévos...",[],NO
5364,https://hdl.handle.net/20.500.11840/pi79059,"Sutton, F.W.",http://www.wikidata.org/entity/Q29043149,"[Frederick William Sutton, Frederick William S...",[],NO
5262,https://hdl.handle.net/20.500.11840/pi78602,"Warden, D.B.",http://www.wikidata.org/entity/Q18122726,"[David Baillie Warden, David Baillie Warden, D...",[],NO
4071,https://hdl.handle.net/20.500.11840/pi66980,Uoya Eikichi (uitgever),http://www.wikidata.org/entity/Q84498310,"[Uoya Eikichi, Uoya Eikichi, 魚屋栄吉, Uoya Eikich...",[],NO
3272,https://hdl.handle.net/20.500.11840/pi63913,Tjokorda Gdé Raka Sukewati,http://www.wikidata.org/entity/Q86819769,[],[],NO
2493,https://hdl.handle.net/20.500.11840/pi60416,"Watei, Ryuho, ...., Seiko, Hazan & (yosegaki)",http://www.wikidata.org/entity/Q86819124,"[Watei, Ryuho, ...., Seiko, Hazan & (yosegaki)...",[],NO


Reason for Mismatch: 
1. Spelling variation of surname: M. (Metin) Ãzlen --> Metin Özlen; Valentyn --> Valentijn
2.  Character encoding went wrong: Shunkôsai Hokushû --> Shunkōsai Hokushū; Seki Yûshi --> Seki Yûshi; Prévost --> Prévost
3. Difficulties in finding surname:
    a. Surname went first like author reference: Warden, D.B. --> David Baillie Warden
    b. not necessary the last word of the name field is surname, sometimes it specifies smth about the person. Anak Agung Gedeh Mereg --> Mereg Anak Agung Gedeh; 
    c. Valentyn V.D.M., F. --> François Valentijn; 
    d. Kobori Sôjitsu Masaharu --> Sôjitsu Masaharu Kobori
4. Group-of people is also considered as instance of human: Watei, Ryuho, ...., Seiko, Hazan & (yosegaki) --> Watei, Ryuho, ...., Seiko, Hazan & (yosegaki) SPL 

- why 4950 and 3272  is empty for wiki_label

In [4]:
df.loc[df["match"] == "YES"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,retrieved_names,match
93,https://hdl.handle.net/20.500.11840/pi20634,J.F.E. (Johan Frederik Engelbert) ten Klooster,http://www.wikidata.org/entity/Q2224964,"[Johan Frederik Engelbert ten Klooster, Johan ...","[Johannes ten Klooster, Johan Frederik Engelbe...",YES
3152,https://hdl.handle.net/20.500.11840/pi63556,D.J. Pannekoek,http://www.wikidata.org/entity/Q86739397,[D.J. Pannekoek],"[D.J. Pannekoek, Mevr. Pannekoek]",YES
2715,https://hdl.handle.net/20.500.11840/pi62383,David Attenborough,http://www.wikidata.org/entity/Q183337,"[David Attenborough, Deivids Atenboro, Devid A...","[Sir David Attenborough, David Attenborough]",YES
5560,https://hdl.handle.net/20.500.11840/pi80139,Dolly Nampijinpa Daniels,http://www.wikidata.org/entity/Q21598135,"[Dolly Nampijinpa Daniels, دولى نامپيچينپا دان...",[Dolly Nampijinpa Daniels],YES
2520,https://hdl.handle.net/20.500.11840/pi60478,Brik de Zwart,http://www.wikidata.org/entity/Q86737184,[Brik de Zwart],"[Brik de Zwart, L. de Zwart]",YES
4458,https://hdl.handle.net/20.500.11840/pi69837,Ann Demeulemeester,http://www.wikidata.org/entity/Q466241,"[Ann Demeulemeester, ann demeulemeester, Ανν Ν...",[Ann Demeulemeester],YES
2146,https://hdl.handle.net/20.500.11840/pi59682,Anzan,http://www.wikidata.org/entity/Q86818445,[Anzan],[Anzan],YES
3424,https://hdl.handle.net/20.500.11840/pi64406,Rose Horse,http://www.wikidata.org/entity/Q86740075,[Rose Horse],"[Rose Horse, Black Pinto Horse]",YES
5616,https://hdl.handle.net/20.500.11840/pi80270,Marlee Napurrula,http://www.wikidata.org/entity/Q82573971,[Marlee Napurrula],"[Mitjili Napurrula, Marlee Napurrula]",YES
1426,https://hdl.handle.net/20.500.11840/pi44598,T. (Tadashi) Nakayama,http://www.wikidata.org/entity/Q7674368,"[Tadashi Nakayama, تاداشي ناكاياما, 中山正]",[Tadashi Nakayama],YES


# Abbreviation Matching

In [5]:
df = pandas.read_pickle("results/NMVWtoWikiAbbreviationMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

3662

In [14]:
df.loc[df["match"] == "NO"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,abbreviations,retrieved_names,match
5111,https://hdl.handle.net/20.500.11840/pi78041,"Sadoon, S.D.H.M.",http://www.wikidata.org/entity/Q83290981,"[S.D.H.M. Sadoon, S.D.H.M. Sadoon, S.D.H.M. Sa...",S. S.D.H.M.,[],NO
4494,https://hdl.handle.net/20.500.11840/pi69976,Wei Changqing,http://www.wikidata.org/entity/Q84499742,"[Changqing Wei, Changqing Wei, Changqing Wei, ...",W. Changqing,[],NO
4538,https://hdl.handle.net/20.500.11840/pi7118,H. (Herman) Stafhell,https://rkd.nl/explore/artists/385437,[],H. (. Stafhell,[],NO
2347,https://hdl.handle.net/20.500.11840/pi6011,C. (Christiaan) Snouck Hurgronje,http://www.wikidata.org/entity/Q731829,"[Snouck Hurgronje, Snouck Hurgronje, Snouck Hu...",C. (. S. Hurgronje,[],NO
1307,https://hdl.handle.net/20.500.11840/pi44178,Dhr. K. (Kommer) Klein,http://www.wikidata.org/entity/Q86736196,[],D. K. (. Klein,[],NO
5751,https://hdl.handle.net/20.500.11840/pi80591,lgnaz Sebastian Klauber,http://www.wikidata.org/entity/Q38073999,"[Ignaz Sebastian Klauber, Ignaz Sebastian Klau...",l. S. Klauber,[],NO
4317,https://hdl.handle.net/20.500.11840/pi69372,G.F.B. (George Felix Bernard) Watrin,http://www.wikidata.org/entity/Q82576123,"[George Felix Bernard Watrin, George Felix Ber...",G. (. F. B. Watrin,[],NO
6097,https://hdl.handle.net/20.500.11840/pi81917,G.M. (Gustaaf Martinus) Oosterling,http://www.wikidata.org/entity/Q64969957,"[جوستاف مارتينوس اوستيرلينج, Gustaaf Martinus ...",G. (. M. Oosterling,[],NO
5389,https://hdl.handle.net/20.500.11840/pi79303,"Brugsch, E.",http://www.wikidata.org/entity/Q73057,"[اميل بروجستش, Émile Charles Albert Brugsch, E...",B. E.,[],NO
955,https://hdl.handle.net/20.500.11840/pi40441,H. (Hamid) Sharifi,http://www.wikidata.org/entity/Q82575034,"[Hamid Sharifi, Hamid Sharifi, Hamid Sharifi]",H. (. Sharifi,[],NO


Reason for Mismatch: 
- Not surname in the last: Sadoon, S.D.H.M. --> S.D.H.M. Sadoon --> S. S.D.H.M.(abbrv.)	
    - Wei Changqing --> Changqing Wei --> Changqing Wei
    - Brugsch, E. --> B. E.
- there is often mention like this: H. (Herman) Stafhell; while abbreviated it went like this,  H. (. Stafhell; A good portion of mismatch happended due to this
- Character encoding went wrong: Niwa Tôkei--> Niwa Tōkei --> N. Tôkei(abbrv.)	

In [6]:
df.loc[df["match"] == "YES"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,abbreviations,retrieved_names,match
3833,https://hdl.handle.net/20.500.11840/pi65949,Emily Nipisa'naaq Alerk,http://www.wikidata.org/entity/Q84495482,[Emily Nipisa'naaq Alerk],E. N. Alerk,[Emily Nipisa'naaq Alerk],YES
2935,https://hdl.handle.net/20.500.11840/pi62868,de Greve,http://www.wikidata.org/entity/Q86819481,[de Greve],d. Greve,[de Greve],YES
3367,https://hdl.handle.net/20.500.11840/pi64219,Mario Grabo,http://www.wikidata.org/entity/Q86740004,[Mario Grabo],M. Grabo,[Mario Grabo],YES
4724,https://hdl.handle.net/20.500.11840/pi76820,"Guinart, J.",http://www.wikidata.org/entity/Q83287135,"[Jose Guinart, José Guinart]",G. J.,[G. J.],YES
5036,https://hdl.handle.net/20.500.11840/pi77764,A. J. C. van Seters,http://www.wikidata.org/entity/Q86741639,"[A. J. C. van Seters, Adriaan Johan Carel van ...",A. J. C. v. Seters,[A. J. C. van Seters],YES
4244,https://hdl.handle.net/20.500.11840/pi69240,L. Hidade,http://www.wikidata.org/entity/Q86741011,[L. Hidade],L. Hidade,[L. Hidade],YES
4577,https://hdl.handle.net/20.500.11840/pi75895,Raden Basoeki Abdullah,http://www.wikidata.org/entity/Q4201116,"[Basoeki Abdullah, Basuki Abdullah, Raden Baso...",R. B. Abdullah,[Raden Basoeki Abdullah],YES
3621,https://hdl.handle.net/20.500.11840/pi65126,Howard Dennis,http://www.wikidata.org/entity/Q84498244,[Howard Dennis],H. Dennis,[Howard Dennis],YES
741,https://hdl.handle.net/20.500.11840/pi39018,Nugroi,http://www.wikidata.org/entity/Q86818040,[Nugroi],Nugroi,[Nugroi],YES
5828,https://hdl.handle.net/20.500.11840/pi80888,R.M. Soebagio,http://www.wikidata.org/entity/Q86742128,[R.M. Soebagio],R. Soebagio,[R.M. Soebagio],YES


# Fuzzy String Matching

In [7]:
df = pandas.read_pickle("results/NMVWtoWikiFuzzyStringMatchResults.pkl")
len(df.loc[df["match"] == "YES"])

6120

In [16]:
df.loc[df["match"] == "NO"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,retrieved_names,match
1283,https://hdl.handle.net/20.500.11840/pi44123,Dhr. S. (Serge) Debecque,http://www.wikidata.org/entity/Q27064216,"[Serge Debecque, Serge Debecque, Serge Debecqu...",[],NO
2245,https://hdl.handle.net/20.500.11840/pi59907,Jôyô Tôkai,http://www.wikidata.org/entity/Q86818680,"[Jôyô Tôkai, Jôyô Tôkai]",[],NO
5192,https://hdl.handle.net/20.500.11840/pi78377,Prof. Dr. K. Florenz,http://www.wikidata.org/entity/Q86741784,[],[],NO
3513,https://hdl.handle.net/20.500.11840/pi64838,V. Mutwa,http://www.wikidata.org/entity/Q84594545,[],[],NO
457,https://hdl.handle.net/20.500.11840/pi37979,W.F. Winckel,http://www.wikidata.org/entity/Q82576026,"[Willem Ferdinand Winckel, Willem Ferdinand Wi...",[],NO
2125,https://hdl.handle.net/20.500.11840/pi59268,P.W.M. Trap,http://www.wikidata.org/entity/Q30304094,"[Pieter Willem Marinus Trap, Pieter Willem Mar...",[],NO
2013,https://hdl.handle.net/20.500.11840/pi57056,prof E.H. Giglioli,http://www.wikidata.org/entity/Q692160,"[Amphiptera, انريكو هيلير جيجليولى, Amphiptera...",[],NO
4854,https://hdl.handle.net/20.500.11840/pi77121,"Solvyns, F.B.",http://www.wikidata.org/entity/Q2682432,"[بالثازار سولڤينس, Balthazar Solvyns, Balthaza...",[],NO
3251,https://hdl.handle.net/20.500.11840/pi63849,jhr mr J.E. de Sturler,http://www.wikidata.org/entity/Q86739702,"[J.E. de Sturler, Jacques Edouard de Sturler, ...",[],NO
535,https://hdl.handle.net/20.500.11840/pi38363,S. (Sıtkı) Olçar,http://www.wikidata.org/entity/Q82574190,"[Sıtkı Olçar, Sıtkı Olçar, Sıtkı Olcar]",[],NO


In [8]:
df.loc[df["match"] == "YES"].sample(20)

,nmvw_uri,name_label,wiki_uri,wiki_label,retrieved_names,match
5087,https://hdl.handle.net/20.500.11840/pi77961,"Meddelkoop, P.",http://www.wikidata.org/entity/Q83286534,"[P. Meddelkoop, Piet Meddelkoop]","[P. Meddelkoop, Piet Meddelkoop, P. Kloos]",YES
2764,https://hdl.handle.net/20.500.11840/pi62484,Bornman,http://www.wikidata.org/entity/Q86819286,[Bornman],"[Bornman, Michiel Boerman, Ank]",YES
3127,https://hdl.handle.net/20.500.11840/pi63485,Nick. Stournaras,http://www.wikidata.org/entity/Q86739339,"[Nick Stournaras, Nikolaos Stournaras, Νικόλαο...","[Nick Stournaras, Arias, Stork]",YES
2017,https://hdl.handle.net/20.500.11840/pi57127,K.W. Gratama,http://www.wikidata.org/entity/Q11299996,"[Koenraad Wolter Gratama, クーンラート・ハラタマ]","[Koenraad Wolter Gratama, Pata]",YES
5016,https://hdl.handle.net/20.500.11840/pi77743,L. Kopper,http://www.wikidata.org/entity/Q86741597,[Lisa Kopper],"[Lisa Kopper, Ellen Klopper, Birgit Maas-Knöpper]",YES
238,https://hdl.handle.net/20.500.11840/pi22627,Willem Frederik Heskes,http://www.wikidata.org/entity/Q2085276,[Wam Heskes],"[Hendrik Hes, Hendrik Frederik Willem Bantje, ...",YES
214,https://hdl.handle.net/20.500.11840/pi22569,Nj. Njoo Djwan Bu,http://www.wikidata.org/entity/Q82574020,[Nj. Njoo Djwan Bu],"[Nj. Njoo Djwan Bu, Njoo Djwan Lian, Oho]",YES
5973,https://hdl.handle.net/20.500.11840/pi81573,J.A. (João Antônio) de Barros,http://www.wikidata.org/entity/Q82570269,[João Antônio de Barros],"[João Antônio de Barros, C.J.N. de Jong, Haan]",YES
4639,https://hdl.handle.net/20.500.11840/pi76565,Vincent Serico,http://www.wikidata.org/entity/Q82575026,[Vincent Serico],"[Vincent Serico, C. Vincenti, Sheri]",YES
179,https://hdl.handle.net/20.500.11840/pi21331,N. (Nicolaas) Pieneman,http://www.wikidata.org/entity/Q720035,"[Nicolaas Pieneman, Николас Пинеман, Նիկոլաս Փ...","[Nicolaas Pieneman, Ank, Pieter Nicolaas van K...",YES


Maybe, a good idea would be to write another code that looks for surname in the whole candiate name. i.e,
- row['LastName'] in candidate.split()

Maybe, it is a good idea to try to seperate title like, Dhr. Mevr. Prof. Dr. ...

Maybe, it is a good idea to detect surname automatically somehow using AI. 

Maybe, we can remove everuthing that falls in bracket (); causing a lot of mismatch for abbreviation match